.github/workflows/docker.yml

```diff
@@ -0,0 +1,84 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - master
+      - next
+      # Add your branch here if you want containers for it
+      - db-WIP
+      - docker-ci
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    name: Build ${{ matrix.config.name }} Container (${{ matrix.metric3d.name }} ; ${{matrix.mpi}})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        mpi: [mpich]
+        metric3d:
+          - name: "With OpenMP"
+            cmake: ON
+            base_prefix: "openmp-"
+            tag_prefix: "3d-"
+          - name: "Without OpenMP"
+            cmake: OFF
+            base_prefix: ""
+            tag_prefix: ""
+        config:
+          - name: "Debug"
+            tag_postfix: "debug"
+            options: "-DCHECK=3"
+            base_postfix: "mini"
+
+          - name: "Optimised"
+            tag_postfix: "opt"
+            options: "-DCHECK=0"
+            base_postfix: "mini"
+
+          - name: "Full"
+            tag_postfix: "full"
+            options: "-DCHECK=3"
+            base_postfix: "full"
```

Comment:
The full would contain `pip`.
The `Debug` is compiled with `-DCHECK=3` as well, so that explains why it is a bit slow.
`opt` could be used for production runs ...